src/PixiEditor/Views/Dialogs/PalettesBrowser.xaml.cs

```diff
@@ -652,14 +647,23 @@ private async void ImportFromFile_OnClick(object sender, RoutedEventArgs e)
 
     private async Task ImportPalette(string fileName, IList<PaletteFileParser> parsers)
     {
-        var parser = parsers.FirstOrDefault(x => x.SupportedFileExtensions.Contains(Path.GetExtension(fileName)));
-        if (parser != null)
+        // check all parsers for formats with same file extension
+        var parserList = parsers.Where(x => x.SupportedFileExtensions.Contains(Path.GetExtension(fileName).ToLower())).ToList();
+
+        if (parserList != null)
         {
-            var data = await parser.Parse(fileName);
+            int index = 0;
+            foreach (var parser in parserList)
+            {
+                var data = await parser.Parse(fileName);
+                index++;
+
+                if (data.IsCorrupted && index == parserList.Count) return; // fail if none of the parsers in our list can read the file
+                if (data.IsCorrupted) continue; // skip to next parser if unable to read
```

Comment:
This might be enough for most cases, however I'd prioritize magic bytes check, if this fails (because some format might not have magic bytes), then I'd fallback to try parsing

Reply:
I'm just relying on the the parser to fail to read the file, which should be checking for magic bytes. I didn't want to change any of the other existing core code. It worked as I tested on several different .pal formats.